_Classes/DeusEx/Classes/DeusExPlayer.uc

```diff
@@ -2399,6 +2397,80 @@ function RecoilEffectTick(float deltaTime)
 	}
 }
 
+// ----------------------------------------------------------------------
+// GetMeleePriority()
+// ----------------------------------------------------------------------
+
+function GetMeleePriority(int damageThreshold)	// Trash: Used to automatically decide what to draw
+{
+	local Inventory anItem;
+	local DeusExWeapon meleeWeapon;
+
+	local DeusExWeapon crowbar, sword, knife, baton, dts;
+
+	For(anItem = Inventory; anItem != None; anItem = anItem.Inventory)	// Go through the entire inventory, check for these melee weapons
+	{
+		if (anItem.IsA('WeaponSword'))
+			sword = DeusExWeapon(anItem);
+		if (anItem.IsA('WeaponCrowbar'))
+			crowbar = DeusExWeapon(anItem);
+		if (anItem.IsA('WeaponCombatKnife'))
+			knife = DeusExWeapon(anItem);
+		if (anItem.IsA('WeaponBaton'))
+			baton = DeusExWeapon(anItem);
+		if (anItem.IsA('WeaponNanoSword'))
+			dts = DeusExWeapon(anItem);
+	}
+
+
+	if (bHardcoreMode)
+	{
+		if (sword != none)	// Prioritize weapons better at breaking containers
+			meleeWeapon = sword;
+		else if (crowbar != none)
+			meleeWeapon = crowbar;
+		else if (knife != none)
+			meleeWeapon = knife;
+		else if (baton != none)
+			meleeWeapon = baton;
+		else if (dts != none)
+			meleeWeapon = dts;
+	}
+	else
+	{
+		if (sword != none && BreaksDamageThreshold(sword, damageThreshold))	// Prioritize weapons based on priority and container DT
```

Comment:
Instead of needing two if blocks, can simply use `if (sword != none && (!bHardcore || BreaksDamageThreshold(sword, damageThreshold))` etc

This should massively reduce the size of this function.